tests/testsuite/jobserver.rs

```diff
@@ -212,33 +215,102 @@ test-runner:
         .env("CARGO", cargo_exe())
         .arg("test-runner")
         .arg("-j2")
-        .with_stderr_contains("[..]this is a runner[..]")
+        .with_stderr_data(str![[r#"
+[FINISHED] `test` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+[RUNNING] unittests src/lib.rs (target/debug/deps/cargo_jobserver_check-[HASH][EXE])
+this is a runner
+
+"#]])
         .run();
 
     // but not from `-j` flag
     p.cargo("run -j2")
         .with_status(101)
-        .with_stderr_contains("[..]no jobserver from env[..]")
+        .with_stderr_data(str![[r#"
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+[RUNNING] `target/debug/cargo-jobserver-check[EXE]`
+thread 'main' panicked at src/main.rs:5:43:
+no jobserver from env: NotPresent
+[NOTE] run with `RUST_BACKTRACE=1` environment variable to display a backtrace
+
+"#]])
```

Comment:
https://github.com/rust-lang/cargo/actions/runs/9803086601/job/27068710146#step:11:4061
```
 running `D:\a\cargo\cargo\target\debug\cargo.exe run -j2`
thread 'jobserver::runner_inherits_jobserver' panicked at tests\testsuite\jobserver.rs:237:10:

---- expected: tests\testsuite\jobserver.rs:229:27
++++ actual:   stderr
   1    1 | [FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
   2    2 | [RUNNING] `target/debug/cargo-jobserver-check[EXE]`
   3    3 | thread 'main' panicked at src/main.rs:5:43:
   4    4 | no jobserver from env: NotPresent
   5    5 | [NOTE] run with `RUST_BACKTRACE=1` environment variable to display a backtrace
        6 + [ERROR] process didn't exit successfully: `target/debug/cargo-jobserver-check[EXE]` ([EXIT_STATUS]: 101)
```
I don't get it why it got passed with or without the error line at local win host, but fails at CI env.